examples/lighting-app/nrfconnect/main/AppTask.cpp

```diff
@@ -99,6 +102,9 @@ int AppTask::Init()
     InitServer();
     PrintQRCode(chip::RendezvousInformationFlags::kBLE);
 
+    // Initialize cluster state to meet with the real light state
+    UpdateClusterState();
```

Comment:
This should happen inside `emberAfPluginOnOffClusterServerPostInitCallback`, presumably, to happen at the right place in the ZCL model layer initialization.  It looks like the `gen` files for lighting-app (and lock-app) don't set up the `CLUSTER_MASK_INIT_FUNCTION` bits for the on/off cluster; they should presumably be regenerated with that bit set, so `emberAfPluginOnOffClusterServerPostInitCallback` will be called.

Reply:
Ok and could you tell me where I can find how to regenerate those files, to check if emberAfPluginOnOffClusterServerPostInitCallback is called and everything works as expected?

Reply:
At the moment you need to open up Simplicity Studio, create a configuration like the one we already have but with the "Cluster server init" box checked under the "Callbacks" tab, then do some hand-editing on the generated code to make it compile in CHIP and add licenses.  And depending on how the version of Simplicity Studio and SDK you are using compares to what was used to generate the files, either hand-edit more stuff or end up with a bunch of extraneous changes.

We're getting close to having a better flow, but we're not there yet.  :(

In the meantime, I created https://github.com/project-chip/connectedhomeip/pull/3180 to change the generated files to ensure the callback is called.  As far as testing, if you just add an `assert(false)` inside `emberAfPluginOnOffClusterServerPostInitCallback` in `examples/lighting-app/lighting-common/gen/callback-stub.c` on top of #3180 it should be obvious whether it's getting called or not.  And then you'd want to remove it from `callback-stub.c` and move it to the relevant apps as needed.

Reply:
I tested it with your changes from #3180 and it works, but I'm not sure about removing this implementation from callback-stub.c. Without removing it, the code obviously will not compile because of the multiple definition, but would you like me to remove it permanently from callback-stub.c and provide empty implementations for all platforms? Wouldn't be better to not remove the implementation, but give it weak attribute?

Reply:
We've had some weird issues with the linker, so are unlikely willing to trust it here.  Ok with you to do the necessary for now and revisit when we think __attribute__(weak) is robust and better enough to justify the work of proving it robust?  our build system for these generated files is still a WIP...

Reply:
In practice, apps really should implement this callback to work properly, right?  I think the right call is to remove from `callback-stub.c`, add empty impls to the apps that won't have a real impl, with TODO comments pointing to issues we file to track implementing those bits.

Reply:
Of course, I was just curious that's why I asked. So I followed your suggestions and I implemented empty callbacks for all platforms support lighting/lock apps. Hope that I did what you mean, but please verify it.